### PR TITLE
Use rule name as cache name

### PR DESCRIPTION
**Pull request by jabolina**
Using the configured rule name as the cache name. I'll update the cache-manager to use the rule name as well.

### CODE REVIEW

**Comment by jabolina** [comment]
Opened https://github.com/gingersnap-project/cache-manager/pull/7 too

**Comment by wburns** [comment]
> I'll update the cache-manager to use the rule name as well.

Don't worry about that, I will take care of that in my new PR https://github.com/gingersnap-project/cache-manager/pull/5 as I ran into this issue because of it :)

File: src/main/java/io/gingersnapproject/cdc/cache/hotrod/HotRodService.java [hunk 1]
```diff
@@ -91,10 +130,10 @@ public OffsetBackend offsetBackendForURI(URI uri) {
       return new HotRodOffsetBackend(remoteCacheManager.getCache(OFFSET_CACHE_NAME));
    }
 
-   private void getOrCreateOffsetBackendCache(RemoteCacheManager rcm) {
+   private void getOrCreateOffsetBackendCache(String name, RemoteCacheManager rcm) {
       rcm.administration()
             .withFlags(CacheContainerAdmin.AdminFlag.VOLATILE)
-            .getOrCreateCache(OFFSET_CACHE_NAME, new StringConfiguration(
+            .getOrCreateCache(name, new StringConfiguration(
```
Comment (wburns) [inline src/main/java/io/gingersnapproject/cdc/cache/hotrod/HotRodService.java:136]: One thing we need to keep in mind that I hadn't thought about is we need to remove an offset when a rule is no longer configured.

File: pom.xml [hunk 2]
```diff
@@ -7,7 +7,7 @@
   <version>0.1-SNAPSHOT</version>
   <properties>
     <compiler-plugin.version>3.8.1</compiler-plugin.version>
-    <maven.compiler.release>11</maven.compiler.release>
+    <maven.compiler.release>17</maven.compiler.release>
```
Comment (wburns) [inline pom.xml:10]: Maybe just put this in a different commit, can still be in the PR

**Review by wburns — APPROVED** [review]
LGTM, just one minor comment about separating out a commit.

File: src/main/java/io/gingersnapproject/cdc/cache/hotrod/HotRodService.java (hunk 1 above)
Comment (wburns) [reply src/main/java/io/gingersnapproject/cdc/cache/hotrod/HotRodService.java:136]: I created https://github.com/gingersnap-project/db-syncer/issues/43 encapsulate all the things we need to remove when a rule is removed

File: pom.xml (hunk 2 above)
Comment (jabolina) [reply pom.xml]: Some PR was merged and updated to 17.

**Comment by jabolina** [comment]
PR rebased.

**Comment by wburns** [comment]
> PR rebased. Actually, better to merge #39 first, as I think we'll need to rebase again.

Actually that one is already conflicted sadly and I would like to get this in so I can update my PR.
